lib/tasks/users.rake

```diff
@@ -149,4 +149,11 @@ namespace :users do
 
     UserPermissionMigrator.migrate(source: source_application, target: target_application)
   end
+
+  desc "Encrypts user OTP keys, temporary task to migrate away from unencrypted keys"
+  task encrypt_otp_keys: :environment do
+    two_factor_users = User.where.not(otp_secret_key: nil)
+
+    two_factor_users.each(&:encrypt)
```

Comment:
I imagine it’d be a bit more efficient to use find_each so the data is loaded in chunks. Not sure if loading them all may exceed memory available